### ISSUE TITLE
acme.sh throws chown error though everything works fine

### ISSUE THREAD

**Issue by IzzySoft** [Neilpang/acme.sh#728]
On issuing a cert or renewing it, I get a bunch of `chown` errors – while the job itself succeeds:

    chown: changing ownership of '/var/www/.well-known': Operation not permitted

(same for all its sub-directories it used). That directory has the correct ownership, and renewal went fine (so maybe that error can be suppressed, too – `chown` fails e.g. if the webroot is owned by `root:root` and you're running the script as `user` – which is perfectly fine if `$webroot/.well-known` is owned by `user`, no `chown` needed then. Could be checked with `if [ "$(stat -c '%U' $_currentRoot/.well-known)" != "$(whoami)"`, for example – or, more thoroughly, see [Bash test if a directory is writable by a given UID?](https://stackoverflow.com/questions/14103806/bash-test-if-a-directory-is-writable-by-a-given-uid)).

This most likely only happens in standalone (or rather "non-apache") mode, [see here for why](https://github.com/Neilpang/acme.sh/blob/master/acme.sh#L3477).

TL;DR: Only do the `chown` if necessary :smile_cat:

**Comment by Neilpang** [comment]
The error message is fixed.

**Comment by IzzySoft** [comment]
Thanks! Will test that ASAP by updating the script. Next domain is due for renewal in a couple of days :smile_cat:
